map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,218 @@
+        String polylineData = IOUtils.toString(getClass().getResourceAsStream("/show_location_history/polyline_data.json"),
+          StandardCharsets.UTF_8);
+        // create a polyline from the location points
+        Polyline routePolyline = (Polyline) Geometry.fromJson(polylineData,
+          SpatialReferences.getWebMercator());
+
+        // create a simulated location data source
+        SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+        // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+        simulatedLocationDataSource.setLocations(
+          routePolyline, new SimulationParameters(Calendar.getInstance(), 30.0, 0.0, 0.0));
+
+        // configure the map view's location display to follow the simulated location data source
+        LocationDisplay locationDisplay = mapView.getLocationDisplay();
+        locationDisplay.setLocationDataSource(simulatedLocationDataSource);
+        locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+        locationDisplay.setInitialZoomScale(7000);
+
+        // create a listener to track the previous location, to draw a route line behind the location display symbol
+        LocationChangedListener locationChangedListener = (LocationDataSource.LocationChangedEvent locationChangedEvent) -> {
+
+          // reset the old polyline connecting the points
+            locationHistoryLineOverlay.getGraphics().clear();
+
+            // add any previous position to the history
+            if (position != null) {
+              // add the new point to the polyline
+              polylineBuilder.addPoint(position);
+              // add the new point to the graphics overlay
+              locationHistoryOverlay.getGraphics().add(new Graphic(position));
+            }
+            // store the current position
+            position = locationChangedEvent.getLocation().getPosition();
+
+            // add the updated polyline to the graphics overlay
+            locationHistoryLineOverlay.getGraphics().add(new Graphic(polylineBuilder.toGeometry()));
+          };
+
+        trackingButton.setOnAction(event -> {
+          // if the user has panned away from the location display, turn it on again
+          if (locationDisplay.getAutoPanMode() == LocationDisplay.AutoPanMode.OFF) {
+            locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+          }
+          // toggle the location tracking when the button is clicked
+          if (isTrackingEnabled) {
+            trackingButton.setText("Start Tracking");
+            simulatedLocationDataSource.removeLocationChangedListener(locationChangedListener);
+          } else {
+            trackingButton.setText("Stop Tracking");
+            simulatedLocationDataSource.addLocationChangedListener(locationChangedListener);
+          }
+          isTrackingEnabled = !isTrackingEnabled;
+        });
```

Comment:
I'm thinking something like the following....but let's have a chat about it as it has its positives and  possible negatives:

```suggestion
          if (trackingButton.isSelected()) {
            trackingButton.setStyle("-fx-background-color: #00ff00; ");
            trackingButton.setText("Tracking Status: ACTIVE");
            simulatedLocationDataSource.addLocationChangedListener(locationChangedListener);
          } else {
            trackingButton.setStyle("-fx-background-color: #F16345; ");
            trackingButton.setText("Tracking Status: STOPPED");
            simulatedLocationDataSource.removeLocationChangedListener(locationChangedListener);
          }
        });
```